### ISSUE TITLE
Addon Validation Error

### ISSUE THREAD

**Issue by hackademix** [mozilla/addons#1295]
[noscript-11.2.1rc2.zip](https://github.com/mozilla/addons/files/5974806/noscript-11.2.1rc2.zip)

Please attach a copy of your add-on, and fill in the following information:

* Uploading a new version self-hosted (beta) of NoScript

* NoScript is listed

* Failed both with web-ext and submitting manually at https://addons.mozilla.org/en-US/developers/addon/noscript/versions/submit/ 

* Validation report: https://addons.mozilla.org/en-US/developers/upload/5bceee1ce54c49f5a1fcf8afa1c4e73d

* ~ 0 AM UTC of Feb the 13th.

**Comment by hackademix** [comment]
Nevermind, I've found the culprit, even though there was no intelligible error message.
I was using the II= operator in nscl/common/tld.js, which apparently makes your validator explode.

**Comment by rpl** [comment]
@hackademix Hi Giogio,
I tried to reproduce this to double-check what kind of validation error we have been showing and if the error is triggered by eslint being unable to parse the source code as I would assume (we are working on making sure to provide some more details when a validation is failing due to a js syntax errors, see mozilla/addons-linter#3556, and so this was also potentially relevant to that work).

I was mainly interested to just see the error as you were experiencing it and so I tried to run the addons-linter (both the current master and an older version I still have installed, version `2.0.1`), but only the older linter version did report a validation error (while the version on master did not report any error at all).

On v. `2.0.1.` I got the following validation error:

```
Validation Summary:

errors          1
notices         0
warnings        0

ERRORS:

Code              Message        Description                      File            Line   Column
JS_SYNTAX_ERROR   JavaScript     There is a JavaScript syntax     background.js   2      5
                  syntax error   error in your code; validation
                                 cannot continue on this file.
```

and the following was the content of background.js:
```
let a = false;
a ||= true;
```

On the older version the error was definitely trigger on the `=` character (which is also confirmed by the "line 2 column 5" part of the validation error).

is this matching the kind (and detail level) of validation error got locally when running `web-ext lint` and when you submitted the addon to AMO?

Based on the different behavior on master and v2.0.1 I think that eslint we are using right now is able to parse the code using `||=` but the older version was throwing a syntax error (but I did't tries to bisect which eslint update did actually stop to fail to parse that js syntax).
